core/src/clients/git.rs

```diff
@@ -463,6 +487,52 @@ impl Git {
         self.run(&["rebase", "--quit"], Opts::default()).await
     }
 
+    // sample output of: git worktree list --porcelain
+    // worktree D:/repos/fellowship
+    // HEAD 6ca1438e074b664470df54319cd6272a4d4d565d
+    // branch refs/heads/f11r-rjd-test3
+    //
+    // worktree D:/repos/fellowship-wt
+    // HEAD b09777a82eaa707f722bcf0d2566b6104f43bc11
+    // detached
+    pub async fn list_worktrees(&self) -> anyhow::Result<Vec<WorktreeInfo>> {
+        let output = self
+            .run_and_collect_output(
+                &["worktree", "list", "--porcelain"],
+                Opts {
+                    ignored_errors: &[],
+                    should_log_stdout: false,
+                    return_complete_error: true,
+                    lfs_mode: LfsMode::Stubs,
+                },
+            )
+            .await?;
+
+        let dir_regex = Regex::new(r"^worktree (.+)").unwrap();
+        let sha_regex = Regex::new(r"^HEAD (.+)").unwrap();
+        let branch_regex = Regex::new(r"^(branch|detached)\s*(.+)?").unwrap();
```

Comment:
Generally with static expressions we've tossed them in a `lazy_static!` though i should say that rust 1.70 has a solution to this that doesn't require a dependency: https://blog.rust-lang.org/2023/06/01/Rust-1.70.0.html#oncecell-and-oncelock

Non-blocking feedback either way.

Reply:
Yeah I can do this